src/transformation/visitors/binary-expression/assignments.ts

```diff
@@ -55,15 +69,16 @@ export function transformAssignment(
         return [arrayLengthAssignment];
     }
 
-    const symbol = ts.isShorthandPropertyAssignment(lhs.parent)
-        ? context.checker.getShorthandAssignmentValueSymbol(lhs.parent)
-        : context.checker.getSymbolAtLocation(lhs);
+    const symbol =
+        lhs.parent && ts.isShorthandPropertyAssignment(lhs.parent)
+            ? context.checker.getShorthandAssignmentValueSymbol(lhs.parent)
+            : context.checker.getSymbolAtLocation(lhs);
 
     const dependentSymbols = symbol ? getDependenciesOfSymbol(context, symbol) : [];
 
-    const left = transformAssignmentLeftHandSideExpression(context, lhs);
+    const left = transformAssignmentLeftHandSideExpression(context, lhs, rightHasPrecedingStatements);
 
-    const rootAssignment = lua.createAssignmentStatement(left, right, lhs.parent);
```

Comment:
Is this changed correct? Wouldn't this 'break' source maps?

Reply:
I changed this temporarily at some point for a reason I can't even remember but it was supposed to be reverted. 😅